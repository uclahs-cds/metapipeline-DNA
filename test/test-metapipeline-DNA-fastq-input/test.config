includeConfig "${projectDir}/config/default.config"
includeConfig "${projectDir}/config/methods.config"
includeConfig "${projectDir}/nextflow.config"

params {
    input_csv = "${projectDir}/test/test-metapipeline-DNA-fastq-input/input.csv"
    output_dir = "${projectDir}/test/output"
    leading_work_dir = "${projectDir}/test/output/leading_tmp_dir" // Should be a /hot/path | Can't use /scratch
    pipeline_work_dir = "/scratch" // For individual pipeline outputs before copying to output_dir
    project_id = 'GZLUCPTA000001'
    save_intermediate_files = false

    partition = 'F72'
    clusterOptions = '--exclusive'
    per_job_cpus = 72
    per_job_memory_GB = 136.8
    max_parallel_jobs = 5
    multi_sample_calling = false

    // Pipeline-specific params under each respective pipeline
    pipeline_params {
        bam2fastq {
            subworkflow_cpus = 8
            get_bam_stats_SAMtools_cpus = 8
            collate_bam_SAMtools_cpus = 8
            // Uncomment to set read group tag values to override that from the original BAMs.
            // This can also be usful if you know your BAM files don't have those tags.
            // sequencing_center =
            // platform_unit =
            // id_for_pu =
        }

        align_DNA {
            enable_spark = true
            mark_duplicates = true
            subworkflow_cpus = 8
            reference_fasta_bwa = '/hot/ref/tool-specific-input/BWA-MEM2-2.2.1/GRCh38-BI-20160721/index/genome.fa'
            aligner = ['BWA-MEM2']
        }

        call_gSNP {
            subworkflow_cpus = 16
            reference_fasta = "/hot/ref/reference/GRCh38-BI-20160721/Homo_sapiens_assembly38.fasta"
            bundle_mills_and_1000g_gold_standard_indels_vcf_gz = "/hot/ref/tool-specific-input/GATK/GRCh38/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz"
            bundle_known_indels_vcf_gz = "/hot/ref/tool-specific-input/GATK/GRCh38/Homo_sapiens_assembly38.known_indels.vcf.gz"
            bundle_v0_dbsnp138_vcf_gz = "/hot/ref/tool-specific-input/GATK/GRCh38/resources_broad_hg38_v0_Homo_sapiens_assembly38.dbsnp138.vcf.gz"
            bundle_hapmap_3p3_vcf_gz = "/hot/ref/tool-specific-input/GATK/GRCh38/hapmap_3.3.hg38.vcf.gz"
            bundle_omni_1000g_2p5_vcf_gz = "/hot/ref/tool-specific-input/GATK/GRCh38/1000G_omni2.5.hg38.vcf.gz"
            bundle_phase1_1000g_snps_high_conf_vcf_gz = "/hot/ref/tool-specific-input/GATK/GRCh38/1000G_phase1.snps.high_confidence.hg38.vcf.gz"
            bundle_contest_hapmap_3p3_vcf_gz = "/hot/ref/tool-specific-input/GATK/GRCh38/Biallelic/hapmap_3.3.hg38.BIALLELIC.PASS.2021-09-01.vcf.gz"
            max_number_of_parallel_jobs = 2
        }

        call_sSNV {
            subworkflow_cpus = 8
            algorithm = ['somaticsniper', 'strelka2', 'mutect2']
            reference = '/hot/ref/reference/GRCh38-BI-20160721/Homo_sapiens_assembly38.fasta'
            exome = false
            call_region = '/hot/ref/tool-specific-input/Strelka2/GRCh38/strelka2_call_region.bed.gz'
            intervals = '/hot/users/czhu/pipeline-germline-somatic/pipeline-germline-somatic/test/test-call-sSNV/intervals.list'
            germline_resource_gnomad_vcf = '/hot/ref/tool-specific-input/GATK/GRCh38/af-only-gnomad.hg38.vcf.gz'
        }

        call_mtSNV {
            subworkflow_cpus = 8
            mt_ref_genome_dir = '/hot/ref/mitochondria_ref/genome_fasta/'
            gmapdb = '/hot/ref/mitochondria_ref/gmapdb/gmapdb_2021-03-08/'
        }
    }
}

methods.set_up()
