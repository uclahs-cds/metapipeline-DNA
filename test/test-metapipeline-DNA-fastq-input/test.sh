#! bin/bash

NXF_WORK=./test/work nextflow run main.nf \
-c test/global.config \
-c test/test-metapipeline-DNA-fastq-input/test.config \
> test/test-metapipeline-DNA-fastq-input/test.out