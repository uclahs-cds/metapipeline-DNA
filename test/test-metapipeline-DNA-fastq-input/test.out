N E X T F L O W  ~  version 21.10.5
Launching `main.nf` [reverent_knuth] - revision: 7ae7e2b7c3
[patient, sample, state, site, index, read_group_identifier, sequencing_center, library_identifier, platform_technology, platform_unit, bam_header_sm, lane, read1_fastq, read2_fastq]
Unknown config attribute `out` -- check config file: /hot/user/hwinata/pipelines/metapipeline-DNA/test/test-metapipeline-DNA-fastq-input/test.config


